posthog/src/main/java/com/posthog/internal/PostHogDecideResponse.kt

```diff
@@ -15,5 +16,44 @@ internal data class PostHogDecideResponse(
     val errorsWhileComputingFlags: Boolean = false,
     val featureFlags: Map<String, Any>?,
     val featureFlagPayloads: Map<String, Any?>?,
+    val flags: Map<String, FeatureFlag>? = null,
     val quotaLimited: List<String>? = null,
+    val requestId: String?,
 ) : PostHogRemoteConfigResponse()
+
+/**
+ * The feature flag data structure for calling the decide API v4. This is what each flag in "flags" will be.
+ * @property key the key of the feature flag
+ * @property enabled whether the feature flag is enabled
+ * @property variant the variant of the feature flag
+ * @property metadata the metadata of the feature flag
+ * @property reason the reason the feature flag was evaluated
+ */
+@IgnoreJRERequirement
```

Comment:
we can stract those to new classes (files), its not common to have multiple classes in a single file on java/kotlin land

Reply:
remember to add those classes that will be serialized/deserialized to the proguard file
https://github.com/PostHog/posthog-android/blob/d269ddb7a1e07590b00d68cd04b62d1b87716554/posthog-android/consumer-rules.pro#L13-L27
GSON uses reflection, and after minification, reflection breaks since classes are renamed, so we need to do this to keep it working when minification is enabled